core/src/main/java/io/stargate/core/metrics/impl/MetricsImpl.java

```diff
@@ -1,14 +1,35 @@
 package io.stargate.core.metrics.impl;
 
 import com.codahale.metrics.MetricRegistry;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.dropwizard.DropwizardExports;
 import io.stargate.core.metrics.api.Metrics;
 
 public class MetricsImpl implements Metrics {
 
   private final MetricRegistry registry;
 
+  private final PrometheusMeterRegistry prometheusMeterRegistry;
+
   public MetricsImpl() {
     registry = new MetricRegistry();
+    prometheusMeterRegistry = initPrometheusMeterRegistry(registry);
+  }
+
+  private PrometheusMeterRegistry initPrometheusMeterRegistry(MetricRegistry metricRegistry) {
+    // note that we are adding the dropwizard exports to the CollectorRegistry that we will pass to
+    // the meter registry
+    DropwizardExports dropwizardExports = new DropwizardExports(metricRegistry);
+    CollectorRegistry collectorRegistry = new CollectorRegistry();
+    collectorRegistry.register(dropwizardExports);
+
+    PrometheusMeterRegistry meterRegistry =
+        new PrometheusMeterRegistry(PrometheusConfig.DEFAULT, collectorRegistry, Clock.SYSTEM);
+    io.micrometer.core.instrument.Metrics.addRegistry(meterRegistry);
```

Comment:
Is this a `static` call? Why do we need it?

Reply:
This static call is to add this registry to the global micrometer composite registry, so that in fact metrics could be reported globally using static methods.. Can remove it in order to force usage only through what `Metrics` exposes?

Reply:
Yeah, I'd like to avoid us relying on `static` variables given that we run in OSGi.

Reply:
OK, will remove for now..